drivers/usb/udc/udc_dwc2.c

```diff
@@ -320,27 +341,43 @@ static int dwc2_tx_fifo_write(const struct device *dev,
 	/* Set number of packets and transfer size */
 	sys_write32((pktcnt << USB_DWC2_DEPTSIZN_PKTCNT_POS) | len, dieptsiz_reg);
 
+	if (priv->bufferdma) {
+		if (!dwc2_dma_buffer_ok_to_use(dev, buf->data, len, cfg->mps)) {
+			/* Cannot continue unless buffer is bounced. Device will
+			 * cease to function. Is fatal error appropriate here?
+			 */
+			return -ENOTSUP;
```

Comment:
`irq_unlock(key);` is missed, but it also slows us down, maybe
```c
	if (IS_ENABLED(CONFIG_ASSERT)) {
		/* __ASSERT inside dwc2_dma_buffer_ok_to_use() */
		dwc2_dma_buffer_ok_to_use(dev, buf->data, len, cfg->mps);
	}
```

Reply:
I will add `irq_unlock(key);`. I don't think the `if (IS_ENABLED(CONFIG_ASSERT)) {` makes any sense here. The idea is that the stack ceases to work (which is rather clear to user) rather than randomly corrupting data (which can be really hard to track).